src/rpc/services/isForkDisputedService/IsForkDisputedRpcMethods.ts

```diff
@@ -0,0 +1,94 @@
+import ARpcMethods from "@/rpc/ARpcMethods";
+import { ATransport } from "@/transport";
+import IsForkDisputedService from "./IsForkDisputedService";
+import { ChannelId, ForkId } from "@/types/types";
+
+class IsForkDisputedRpcMethods extends ARpcMethods {
+    service: IsForkDisputedService;
+
+    constructor(transport: ATransport, service: IsForkDisputedService) {
+        super(transport, service.p2pManager);
+        this.service = service;
+    }
+
+    /**
+     * Peer receives dispute acknowledgment request
+     * Check if fork is disputed and respond accordingly
+     */
+    public async onDisputeAcknowledgmentRequest(
+        channelId: ChannelId,
+        forkId: ForkId
+    ) {
+        console.log(
+            `Received dispute acknowledgment request for fork ${forkId}`
+        );
+
+        // Check if fork is disputed locally
+        const isDisputedLocal =
+            await this.p2pManager.stateManager.diamondStateMachine.localDiamondContract.isForkDisputed(
+                channelId,
+                forkId
+            );
+
+        if (isDisputedLocal) {
+            console.log(
+                `Fork ${forkId} is disputed on local diamond, responding`
+            );
+            const senderTransport = this.senderTransport;
+            if (!senderTransport) {
+                return;
+            }
+            return this.service.respondToDisputeAcknowledgment(
+                senderTransport,
+                channelId,
+                forkId
+            );
+        }
+
+        // Check if fork is disputed on chain
+        const isDisputedOnChain =
+            await this.p2pManager.stateManager.stateChannelManagerContract.isForkDisputed(
+                channelId,
+                forkId
+            );
+
+        if (isDisputedOnChain) {
+            console.log(`Fork ${forkId} is disputed on-chain, responding`);
+            const senderTransport = this.senderTransport;
+            if (!senderTransport) {
+                return;
+            }
+            return this.service.respondToDisputeAcknowledgment(
+                senderTransport,
+                channelId,
+                forkId
+            );
```

Comment:
cleaner to do this:
```typescript
isDisputed = local.isForkDisputed...
if(!isDisputed)isDisputed = onChain.isForkDisputed...
if(isDisputed){
    respond()
    return
}
//else
disconnectAndBlacklist
```
That way you don't have to keep track of 2 reponse paths, just 1